src/transformers/training_args.py

```diff
@@ -496,7 +503,21 @@ def _setup_devices(self) -> Tuple["torch.device", int]:
         else:
             # Here, we'll use torch.distributed.
             # Initializes the distributed backend which will take care of synchronizing nodes/GPUs
-            torch.distributed.init_process_group(backend="nccl")
+            #
+            # deepspeed performs its own DDP internally, and requires the program to be started with:
+            # deepspeed  ./program.py
+            # rather than:
+            # python -m torch.distributed.launch --nproc_per_node=2 ./program.py
+            if self.deepspeed:
+                from .integrations import is_deepspeed_available
```

Comment:
It's safe to import this at the top of the file, I think.

Reply:
```
ImportError: cannot import name 'is_deepspeed_available' from partially initialized module 'transformers.integrations' (most likely due to a circular import) (/mnt/nvme1/code/huggingface/transformers-deepspeed/src/transformers/integrations.py)
```

Reply:
Ah yes, integrations depends on trainer_callback which then depends on training_arguments, so this can't work. We could mode `is_deepspeed_available` in `file_utils` to make this cleaner.

Reply:
it's an integration so it best belongs where it is. But I'm fine either way. You pick.

Reply:
Let's keep it here then.